genienlp/tasks/generic_dataset.py

```diff
@@ -1929,3 +1929,52 @@ def return_splits(cls, name, path='.data', train='train', validation='dev', test
         return Split(train=all_train_data, eval=all_validation_data, test=all_test_data), Split(
             train=train_path, eval=validation_path, test=test_path
         )
+
+
+class OODDataset(CQA):
+    name = 'ood'
+
+    def __init__(self, path, lower=False, cached_path=None, skip_cache=False, **kwargs):
+        examples = []
+        labels = {'neg': 'negative', 'pos': 'positive'}
+        question = 'Is this sentence in-domain or out-domain?'
+
+        cache_name = os.path.join(cached_path, os.path.basename(path))
+        if os.path.exists(cache_name) and not skip_cache:
+            logger.info(f'Loading cached data from {cache_name}')
+            examples = torch.load(cache_name)
+        else:
+            for label in ['pos', 'neg']:
+                for fname in glob.iglob(os.path.join(path, label, '*.tsv')):
+                    with open(fname, 'r') as f:
+                        line = f.readline()
+                        while line:
+                            context = line.split('\t')[2]
+                            answer = labels[label]
+                            examples.append(Example.from_raw(
+                                make_example_id(self, len(examples)),
```

Comment:
You actually have the example ID in the file already, you don't need to construct it dynamically. This construction seems to depend on the globbing order for files which might not be reliable.